### PR TITLE
disable building rodeos and eosio.tester

### PR DESCRIPTION
**Pull request by linhuang-blockone**
## Change Description

This PR disables building `rodeos` and `eosio-tester`. The main goal is to remove dependencies on `abieos` so that we can build `eosio` with more than just `clang` and `g++ 8` (currently it fails with other flavors).

After disabling rodeos and eosio-tester, it has been verified that `eosio` can be built with `g++ 9` and `g++ 10`.

The changes in `db_combined.hpp` is for fixing multiple definition link error found by `g++ 10`

## Change Type
**Select ONE**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: libraries/chain/include/eosio/chain/backing_store/db_combined.hpp [hunk 1]
```diff
@@ -207,7 +207,10 @@ class rocksdb_whole_db_table_collector {
          all_secondary_indices_;
 };
 
-auto process_all = []() { return true; };
+// This used to be "auto process_all = []() { return true; };"
+// g++10 correctly reports a link error of multiple definitions
+// as db_combined.hpp is included by mutiple source files.
```
Comment (heifner) [inline libraries/chain/include/eosio/chain/backing_store/db_combined.hpp:212]: I don't see that this comment adds anything. I would say remove it.

File: libraries/CMakeLists.txt [hunk 2]
```diff
@@ -52,8 +52,6 @@ set_property(GLOBAL PROPERTY CTEST_CUSTOM_TESTS_IGNORE
   yubico_otp wrap_data wrap info import_rsa import_authkey generate_hmac generate_ec\
   attest pbkdf2 parsing ${_CTEST_CUSTOM_TESTS_IGNORE}")
 
-add_subdirectory( rodeos )
```
Comment (heifner) [inline libraries/CMakeLists.txt:55]: Can we make it a cmake option to enable rodeos and eosio-tester?

**Review by heifner — CHANGES_REQUESTED** [review]

File: libraries/chain/include/eosio/chain/backing_store/db_combined.hpp (hunk 1 above)
Comment (larryk85) [reply libraries/chain/include/eosio/chain/backing_store/db_combined.hpp:212]: I agree, the comment is frivolous.

File: libraries/CMakeLists.txt (hunk 2 above)
Comment (larryk85) [reply libraries/CMakeLists.txt:55]: Seems kind of heavy weight.  It currently is out of sync and needs to be updated to work correctly, so toggling it with an option is a lie to the end user.

**Review by larryk85 — APPROVED** [review]

File: libraries/chain/include/eosio/chain/backing_store/db_combined.hpp (hunk 1 above)
Comment (linhuang-blockone) [reply libraries/chain/include/eosio/chain/backing_store/db_combined.hpp]: Removed.

File: libraries/CMakeLists.txt (hunk 2 above)
Comment (heifner) [reply libraries/CMakeLists.txt:55]: If that broken, then shouldn't we remove the directories out of the branch for now?

**Review by heifner — APPROVED** [review]

**Pull request merged**